plan/expression_rewriter.go

```diff
@@ -67,13 +67,26 @@ func (b *planBuilder) rewrite(expr ast.ExprNode, p LogicalPlan, aggMapper map[*a
 // er.preprocess(expr), which returns a new expr. Then we use the new expr in `Leave`.
 func (b *planBuilder) rewriteWithPreprocess(expr ast.ExprNode, p LogicalPlan, aggMapper map[*ast.AggregateFuncExpr]int, asScalar bool, preprocess func(ast.Node) ast.Node) (
 	expression.Expression, LogicalPlan, error) {
-	er := &expressionRewriter{
-		p:          p,
-		aggrMap:    aggMapper,
-		b:          b,
-		asScalar:   asScalar,
-		ctx:        b.ctx,
-		preprocess: preprocess,
+	b.rwStackPointer++
+	var er *expressionRewriter
+	if len(b.rewriterStack) < b.rwStackPointer {
+		er = &expressionRewriter{
+			p:          p,
+			aggrMap:    aggMapper,
+			b:          b,
+			asScalar:   asScalar,
+			ctx:        b.ctx,
+			preprocess: preprocess,
+		}
+		b.rewriterStack = append(b.rewriterStack, er)
```

Comment:
Will this cause too many growslice in some scenario?

Reply:
In most cases, there won't be many rewriters.